df-valve/src/main/java/com/hashmapinc/tempus/witsml/valve/dot/GraphQLQueryConverter.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright © 2018-2018 Hashmap, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hashmapinc.tempus.witsml.valve.dot;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hashmapinc.tempus.WitsmlObjects.AbstractWitsmlObject;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.json.JsonArray;
+import java.io.IOException;;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+//This class takes a WITSML Query search and converts it to a GraphQL search
+class GraphQLQueryConverter {
```

Comment:
Test the fuck out of this class. We'll need to add a card for that. We'll want to make sure we cover all version/type permutations.